[Neo Sun Han] iP

neosunhan opened the pull request:
# Duke

> "Best app ever!" – a satisfied user

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download duke.jar from [here](https://github.com/neosunhan/ip/releases/tag/v0.1)
2. copy the jar file into an empty folder
3. open a command window in that folder
4. run the command `java -jar duke.jar`

And it is **FREE**! :+1:

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Prerequisites: JDK 11

## Using the `Duke` class

The `Duke` constructor takes in a `java.nio.file.Path` object that represents
the path to the storage file (the text file used to store tasks). Users can then
call the `run` method to start Duke.

```java
Path storagePath = Paths.get(System.getProperty("user.dir"), "data", "duke.txt");
new Duke(storagePath).run();
```

Commit:
Add Gradle support

neosunhan committed:
Implement greet, echo, exit functionality

neosunhan committed:
Add the ability to store and display text entered by user

neosunhan committed:
Add ability to mark tasks as done and not done

neosunhan committed:
Add support for ToDos, Events, Deadlines

neosunhan committed:
Automate Text UI Testing

neosunhan committed:
Add error-handling

neosunhan committed:
Add support for deleting tasks

neosunhan committed:
Save tasks in the hard disk whenever the task list changes

neosunhan committed:
Store dates as  objects

neosunhan committed:
Merge branch 'branch-Level-7'

Saves tasks in hard disk automatically whenever the task list changes.
Loads data from hard disk when Duke starts up.

neosunhan committed:
Merge branch 'branch-Level-8'

Represent dates as `LocalDate` objects.

neosunhan committed:
Account for tasks in hard disk during regression testing

neosunhan committed:
Add `Ui`, `Storage`, `Parser` and `Command` classes

neosunhan committed:
Add ui, storage, parser packages

neosunhan committed:
Merge branch 'add-gradle-support' of github.com:nus-cs2103-AY2223S1/ip

neosunhan committed:
Configure checkstyle

neosunhan committed:
Add JUnit Tests

neosunhan committed:
Update mainClassName

neosunhan committed:
Add JavaDoc

neosunhan committed:
Add punctuation to JavaDoc params

neosunhan committed:
Add Find command

neosunhan committed:
Merge branch 'branch-Level-9'

neosunhan committed:
Add JavaDoc for Find functionality

neosunhan committed:
Implement basic GUI

neosunhan committed:
Implement GUI for Duke

neosunhan committed:
Delete text-based UI

neosunhan committed:
Refactor GUI components into ui package

neosunhan committed:
Remove unnecessary use of this keyword

neosunhan committed:
Refactor xml files

neosunhan committed:
Merge branch 'branch-Level-10'

xav168 commented on src/main/java/duke/command/AddEventCommand.java:
```diff
@@ -0,0 +1,36 @@
+public class AddEventCommand extends Command {
+    public static final String COMMAND_WORD = "event";
+
+    private final String desc;
```
hmm shouldn't desc be DESC since its a constant?

xav168 commented on src/main/java/duke/command/AddEventCommand.java:
```diff
@@ -0,0 +1,36 @@
+    public static final String COMMAND_WORD = "event";
+
+    private final String desc;
+    private final LocalDate at;
```
similarly, shouldn't at be AT?

xav168 commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,36 @@
+/**
+ * Command to add a Deadline to the list of tasks.
+ */
+public class AddDeadlineCommand extends Command {
```
hmm perhaps could consider a different name for the class.

xav168 reviewed (commented):
LGTM, just a few nits here and there

neosunhan committed:
Add assertions

neosunhan committed:
Refactor boolean for task index

neosunhan committed:
Refactor duke welcome message

neosunhan committed:
Delete unused file

neosunhan committed:
Refactor storage strings

neosunhan committed:
Merge pull request #1 from neosunhan/branch-A-Assertions

Add assertions

neosunhan committed:
Merge branch 'master' into branch-A-CodeQuality

neosunhan committed:
Merge pull request #2 from neosunhan/branch-A-CodeQuality

Code Quality fixes

neosunhan committed:
Setup CI

neosunhan committed:
Fix formatting

Several lines in Deadline.java and Event.java are more than 120
characters long.

Breaking these lines into 2 lines improves readability.

neosunhan committed:
Fix string representation for tasks

String representation of tasks is incorrect, causing it to fail test
cases in the CI.

Refactoring done in 0db19f introduced a bug where the string
representation of tasks had a whitespace character removed, thus causing
regression tests to fail.

This commit fixes the bug by adding back the whitespace character.

neosunhan committed:
Implement mass operations

The delete, mark and unmark commands all take 1 task index as an
argument.

Performing the same operation on multiple items requires many commands,
and is slow.

Adding the ability to specify multiple task indices as as the arguments
for delete, mark and unmark commands will allow the user to use the app
more efficiently.

neosunhan committed:
Update tests for Parser

neosunhan committed:
Add name to main window

neosunhan committed:
Anchor send button to bottom of window

neosunhan committed:
Add representative screenshot

neosunhan committed:
Report number of tasks loaded from storage

neosunhan committed:
Update user guide

neosunhan committed:
Update UI screenshot

neosunhan committed:
Refactor `running`

neosunhan committed:
Fix javadoc

neosunhan committed:
Merge pull request #4 from neosunhan/3-code-quality

Code Quality fixes

neosunhan committed:
Add Javadoc

neosunhan committed:
Update User Guide

neosunhan committed:
Update User Guide

neosunhan committed:
Update JavaDoc

neosunhan committed:
Remove trailing whitespace
